src/core/Akka.Remote/Transport/Helios/HeliosTcpTransport.cs

```diff
@@ -110,7 +110,9 @@ protected override void OnMessage(NetworkData data, IConnection responseChannel)
         /// <param name="erroredChannel">The handle to the Helios channel that errored.</param>
         protected override void OnException(Exception ex, IConnection erroredChannel)
         {
-            ChannelLocalActor.Notify(erroredChannel, new UnderlyingTransportError(ex, "Non-fatal network error occurred inside underlying transport"));
+            // Want to notify only for encoding exceptions
+            if(!(ex is HeliosConnectionException))
+                ChannelLocalActor.Notify(erroredChannel, new UnderlyingTransportError(ex, "Non-fatal network error occurred inside underlying transport"));
```

Comment:
Just taking a stab in the dark here, but aren't we interested in any "fatal" error too? 
Are we getting those exceptions in some other observable way?